src/content/app/tools/blast/components/blast-species-selector/BlastSpeciesSelectorHeader.tsx

```diff
@@ -37,43 +33,35 @@ const BlastSpeciesSelectorHeader = (props: Props) => {
 
   const selectedSpeciesList = useSelector(getSelectedSpeciesList);
 
-  const onSwitchToSequence = () => {
-    dispatch(switchToSequencesStep());
-  };
-
   const onClearAll = () => {
     dispatch(clearSelectedSpecies());
   };
 
+  const headerClass = classNames(styles.header, {
+    [styles.smallScreenHeader]: compact
+  });
+
   return (
-    <div className={styles.header}>
+    <div className={headerClass}>
       <div className={styles.headerGroup}>
         <span className={styles.headerTitle}>Blast against</span>
         <span className={styles.speciesCounter}>
           {selectedSpeciesList.length}
         </span>
         <span className={styles.maxSpecies}>of 7 species</span>
-        {compact && (
-          <span className={styles.clearAll} onClick={onClearAll}>
-            Clear all
-          </span>
-        )}
       </div>
-      <div className={styles.headerGroup}>
-        {!compact && (
+      {compact && (
+        <span className={styles.clearAll} onClick={onClearAll}>
+          Clear all
+        </span>
+      )}
+      {!compact && (
+        <div className={styles.headerGroup}>
```

Comment:
what's the purpose of this wrapper for a single "Clear all" element?

Reply:
My bad, i missed cleaning this one up and changing things around. I have pushed the changes now.